### PR TITLE
Make winml_test_common free of test macros

### PR DESCRIPTION
**Pull request by ryanlai2**
This change makes winml_test_common free of any test macros so that this static library can be used across different testing frameworks.

### CODE REVIEW

File: cmake/winml_unittests.cmake [hunk 1]
```diff
@@ -59,10 +59,11 @@ add_dependencies(winml_test_common
   winml_api
   winml_dll
 )
-target_compile_definitions(winml_test_common PRIVATE BUILD_GOOGLE_TEST)
-set_winml_target_properties(winml_test_common)
 
+set_winml_target_properties(winml_test_common)
 file(GLOB winml_test_api_src CONFIGURE_DEPENDS "${WINML_TEST_SRC_DIR}/api/*.cpp")
+# set global variable for the api test source list for other test cmake files to use
+set(winml_test_api_src_list ${winml_test_api_src} PARENT_SCOPE)
```
Comment (smk2007) [inline cmake/winml_unittests.cmake]: >PARENT_SCOPE [](start = 50, length = 12)

worried that this is a little brittle as it is dependent on the include structure of cmakelist.txt.

I believe functions are global scoped - can we try getting the list of files with an out param. 

the cppwinrt sdk helpers have examples of out params in cmake.

File: winml/test/common/SqueezeNetValidator.cpp [hunk 2]
```diff
@@ -261,7 +282,7 @@ void ModelValidator::SqueezeNet(
           std::stringstream ss;
           ss << "EXPECTED: " << outDataExpected.GetAt(i) << " , ACTUAL: " << outDataActual.GetAt(i)
                 << "instance " << instance << ", element " << i;
-          WINML_LOG_ERROR(ss.str().c_str());
+          throw winrt::hresult_invalid_argument(std::wstring_convert<std::codecvt_utf8<wchar_t>>().from_bytes(ss.str()));
```
Comment (smk2007) [inline winml/test/common/SqueezeNetValidator.cpp]: >invalid_argument [](start = 31, length = 16)

i believe that this is "unexpected"

Comment (smk2007) [reply winml/test/common/SqueezeNetValidator.cpp]: same in other places.

---
In reply to: [370428403](https://github.com/microsoft/onnxruntime/pull/2902#discussion_r370428403) [](ancestors = 370428403)

File: winml/test/common/googletest/main.cpp [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #include <iostream>
 #include <unordered_map>
-#include <gtest/gtest.h>
+#include "test.h"
```
Comment (smk2007) [inline winml/test/common/googletest/main.cpp]: >test [](start = 10, length = 4)

should always be gtest.h!

File: cmake/winml_unittests.cmake (hunk 1 above)
Comment (ryanlai2) [reply cmake/winml_unittests.cmake]: sounds good!

File: cmake/winml_unittests.cmake [hunk 4]
```diff
@@ -36,7 +36,7 @@ function(add_winml_test)
     list(REMOVE_DUPLICATES _UT_DEPENDS)
   endif()
 
-  add_executable(${_UT_TARGET} ${_UT_SOURCES})
+  add_executable(${_UT_TARGET} ${_UT_SOURCES} ${WINML_TEST_SRC_DIR}/common/googletest/main.cpp)
```
Comment (tiagoshibata) [inline cmake/winml_unittests.cmake]: We weren't building the same file more than once. Adding `googletest/main.cpp` to each test target will build it multiple times, can you move it into one of the reusable static libraries, or create one for it?

File: winml/test/common/googletest/main.cpp [hunk 5]
```diff
@@ -0,0 +1,44 @@
+#include "testPch.h"
```
Comment (tiagoshibata) [inline winml/test/common/googletest/main.cpp]: How did you arrive at the new formatting, did you run `clang-format` on the file? Just curious if you used it or fixed it by hand.

Comment (ryanlai2) [reply winml/test/common/googletest/main.cpp]: Yep! clang-format was ran in VS.

File: cmake/winml_unittests.cmake [hunk 6]
```diff
@@ -43,14 +43,36 @@ function(add_winml_test)
...
 endfunction()
 
+function(get_winml_test_scenario_src
+  lotus_root_dir
+  output_winml_test_scenario_src
+)
+  set(WINML_TEST_SRC_DIR ${lotus_root_dir}/winml/test)
+  if (onnxruntime_USE_DML)
+    file(GLOB winml_test_scenario_src CONFIGURE_DEPENDS "${WINML_TEST_SRC_DIR}/scenario/cppwinrt/*.cpp")
+    set(winml_test_scenario_libs "onnxruntime_providers_dml" PARENT_SCOPE)
```
Comment (smk2007) [inline cmake/winml_unittests.cmake]: >winml_test_scenario_libs [](start = 8, length = 24)

this needs to be an out param no?

File: cmake/winml_unittests.cmake [hunk 7]
```diff
@@ -43,14 +43,36 @@ function(add_winml_test)
...
     COMMAND ${_UT_TARGET}
     WORKING_DIRECTORY $<TARGET_FILE_DIR:${_UT_TARGET}>
   )
 endfunction()
 
+function(get_winml_test_scenario_src
+  lotus_root_dir
+  output_winml_test_scenario_src
+)
+  set(WINML_TEST_SRC_DIR ${lotus_root_dir}/winml/test)
```
Comment (smk2007) [inline cmake/winml_unittests.cmake]: >WINML_TEST_SRC_DIR [](start = 6, length = 18)

can we use this from the outer scope? it looks like it is references in other functions?

File: cmake/winml_unittests.cmake [hunk 8]
```diff
@@ -43,14 +43,36 @@ function(add_winml_test)
...
+    set(winml_test_scenario_src "${WINML_TEST_SRC_DIR}/scenario/cppwinrt/scenariotestscppwinrt.cpp")
+  endif()
+  set(${output_winml_test_scenario_src} ${winml_test_scenario_src} PARENT_SCOPE)
+endfunction()
+
+function(get_winml_test_api_src
+  lotus_root_dir
+  output_winml_test_api_src
+)
+  set(WINML_TEST_SRC_DIR ${lotus_root_dir}/winml/test)
```
Comment (smk2007) [inline cmake/winml_unittests.cmake]: >WINML_TEST_SRC_DIR [](start = 6, length = 18)

same...

**Pull request merged**